### PR TITLE
more spec v0.3.0 adjustments: Transfer processing, some epoch processing changes, and sum_effective_balance/get_total_balance cleanup

### PR DESCRIPTION
**Pull request by tersec**

### CODE REVIEW

File: beacon_chain/spec/validator.nim [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2018 Status Research & Development GmbH
+# Copyright (c) 2019 Status Research & Development GmbH
```
Comment (mratsim) [inline beacon_chain/spec/validator.nim]: Maybe just

`Copyright (c) 2018-Present Status Research & Development GmbH`

Comment (stefantalpalaru) [reply beacon_chain/spec/validator.nim]: It's usually specified in this form: "2018-2019".

Comment (tersec) [reply beacon_chain/spec/validator.nim]: Done. I went with @stefantalpalaru's suggestion.

File: beacon_chain/spec/beaconstate.nim [hunk 2]
```diff
@@ -365,15 +364,15 @@ proc checkAttestation*(
       attestation_slot = attestation.data.slot, state_slot = state.slot)
     return
 
+  # Can't underflow, because GENESIS_SLOT > MIN_ATTESTATION_INCLUSION_DELAY
```
[the remark this answers is not in the document]
Comment (tersec) [reply beacon_chain/spec/beaconstate.nim]: Assertion added.

**Pull request merged**